### PR TITLE
[CI] Install Codecov on [`deb`,`rpm`] x [`amd`,`arm`] images

### PR DESCRIPTION
**Pull request by amenasria**
We want to start running codecov on every OS, so I'm adding codecov on all images running the go tests.

### CODE REVIEW

File: setup_codecov.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -ex
+
+CODECOV_VERSION=0.6.1
+
+# Integrity checking the uploader
+curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --no-default-keyring --keyring trustedkeys.gpg --import
```
Comment (spencergilbert) [inline setup_codecov.sh:8]: TIL keybase still exists.

**Review by spencergilbert — APPROVED** [review]

**Review by FlorentClarret — APPROVED** [review]
LGTM!

File: setup_codecov.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -ex
```
Comment (chouetz) [inline setup_codecov.sh:3]: the `x` is risky, isn't it? Why not the usual 
```suggestion
set -euo pipefail
```

**Review by chouetz — APPROVED** [review]

File: setup_codecov.sh (hunk 2 above)
Comment (amenasria) [reply setup_codecov.sh:3]: Good point !

**Pull request merged**